react/week2/todolist/src/App.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from "react";
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [
+        {
+          id: 1,
+          description: "Get out of bed",
+        },
+        {
+          id: 2,
+          description: "Brush teeth",
+        },
+        {
+          id: 3,
+          description: "Eat breakfast",
+        },
+      ],
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>todo App</h1>
+        <button onClick={this.addTodoItem}>Add Todo</button>
+        <ul>
+          {this.state.todos.map((task, i) => {
+            return (
+              <li key={i}>
+                {task.description}{" "}
+                <button data-index={i} onClick={this.deleteTodoItem}>
+                  Completed
+                </button>
+              </li>
+            );
+          })}
+        </ul>
+      </div>
+    );
+  }
+
+  addTodoItem = () => {
+    const updatedTodo = this.state.todos;
+    updatedTodo.push({
+      id: Math.floor(Math.random() * 100),
+      description:
+        updatedTodo[Math.floor(Math.random() * updatedTodo.length)].description,
+    });
```

Comment:
What if `this.state.todos` was empty? Then you get an error because `updatedTodo` wouldn't contain any items and you can't get `desciption` of an undefined.
I would recomend to finde another way to add a todo